toolchain/sem_ir/facet_type_info.cpp

```diff
@@ -32,6 +34,13 @@ static auto RewriteLess(const FacetTypeInfo::RewriteConstraint& lhs,
          std::tie(rhs.lhs_id.index, rhs.rhs_id.index);
 }
 
+// Canonically ordered by the numerical ids.
+static auto NamedConstraintsLess(const SpecificNamedConstraint& lhs,
+                                 const SpecificNamedConstraint& rhs) -> bool {
+  return std::tie(lhs.named_constraint_id.index, lhs.specific_id.index) <
+         std::tie(rhs.named_constraint_id.index, rhs.specific_id.index);
+}
```

Comment:
It looks like there's already operator== on `SpecificNamedConstraint`, had you considered doing by adding operator< or operator<=> to the type? I suppose that'd really be best if you did the same for `SpecificInterface` (because then you could maybe write `a < b` instead of `compare(a, b)`) -- is that an issue?

Reply:
Generally IDs should not be treated as ordered. Which means a struct of IDs should not be either. Said another way, I don't think we should add operator< to SpecificNamedConstraint (or SpecificInterface) if we wouldn't add them to all the ID types, which we don't want to. This is an implementation detail of canonicalization, producing a consistent but arbitrary ordering.